### ISSUE TITLE
Cassandra_role - Logs passwords when passed via loop

### ISSUE THREAD

**Issue by tlb1galaxy** [ansible-collections/community.cassandra#125]
##### SUMMARY

Using the Cassandra_role module.  When enter the "password" or "login_password" field directly into the task, the output will respect the 'VALUE_SPECIFIED_IN_NO_LOG_PARAMETER'.
When the same fields are passed via loop (encrypted or plain text), the log output will expose the password.
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

cassandra_role
##### ANSIBLE VERSION

```paste below
ansible 2.10.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/etc/ansible/library/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.8 (default, Aug 24 2020, 17:57:11) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]
```

##### CONFIGURATION

```paste below
COLLECTIONS_PATHS(/etc/ansible/ansible.cfg) = ['/etc/ansible/collections']
DEFAULT_CALLBACK_WHITELIST(/etc/ansible/ansible.cfg) = ['ansible.posix.profile_tasks']
DEFAULT_HOST_LIST(/etc/ansible/ansible.cfg) = ['/etc/ansible']
DEFAULT_MODULE_PATH(/etc/ansible/ansible.cfg) = ['/etc/ansible/library/modules']
DEFAULT_MODULE_UTILS_PATH(/etc/ansible/ansible.cfg) = ['/etc/ansible/library/modules_utils']
DEFAULT_TRANSPORT(/etc/ansible/ansible.cfg) = smart
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False

```

##### OS / ENVIRONMENT

Host: CentOS7 - 4.18.0-240.10.1.el8_3.x86_64
Target: same as above

##### STEPS TO REPRODUCE

Send fields 'name', 'password' via loop to the task

```yaml
- name: cassandra_roles_create_admin
  community.cassandra.cassandra_role:
    name: "{{ item.name }}"
    state: present
    login: "{{ item.login }}"
    super_user: "{{ item.super_user }}"
    login_host: "{{ ansible_env.SSH_CONNECTION.split(' ')[2] }}"
    login_user: "cassandra"
    login_password: "{{ item.login_password }}"
  vars:
    ansible_python_interpreter: /usr/bin/python3
  run_once: true
  when: ansible_fqdn == ansible_play_batch[0]
  loop: "{{ var_cassandra_roles }}"

# ./group_vars/cassandra_cluster.yml
var_cassandra_roles:
  - { name: user1, password: "pasword456", login: yes, super_user: yes, login_password: "cassandra" }
  - { name: user2, password: "{{ var_user2_pass }}", login: yes, super_user: yes, login_password: "cassandra" }

var_user2_pass: !vault |
          $ANSIBLE_VAULT;1.1;AES256
          3537313634346239XXXXXXXXXXXXXXXXXXXXXXX
# encrypted via -> ansible-vault encrypt_string 'secret_data' --name 'my_secret'
```

##### EXPECTED RESULTS
```
<!--- Describe what you expected to happen when running the steps above -->
 "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "login_port": 9042,
            "login_user": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "name": "********admin",
            "options": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
```

##### ACTUAL RESULTS
```
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->
changed: [testserver08.tlb1.thinkon.net] => (item={'name': 'user2', 'password': 'password999', 'login': True, 'super_user': True, 'login_password': 'cassandra'}) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cql": "CREATE ROLE user2 WITH SUPERUSER = True AND LOGIN = True AND PASSWORD = '********' ",
    "invocation": {
        "module_args": {
            "data_centers": null,
            "debug": false,
            "keyspace_permissions": null,
            "login": true,
            "login_host": [
                "1.1.1.1"
            ],
            "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "login_port": 9042,
            "login_user": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "name": "user2",
            "options": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "roles": null,
            "state": "present",
            "super_user": true
        }
    },
    "item": {
        "login": true,
        "login_password": "cassandra",
        "name": "user2",
        "password": "password999",
        "super_user": true
    },
    "role": "user2"
```

```paste below

```

**Comment by rhysmeister** [comment]
Hello,

This is an issue with Ansible itself. I don't have any control over the loop when used in with_items. Some discussion here...

https://groups.google.com/g/ansible-project/c/SaAOoU7g-AU

Suggestions are basically....

1. Don't loop over sensitive items.
2. Use no_log: True at the module level.


Cheers,

Rhys

**Comment by tlb1galaxy** [comment]
Hello Rhys,
Thanks for the quick response.  I tested the same logic with the 'ansible.builtin.user' module to see if it was a loop condition limitation or module.  Since I saw that module respected the 'No_log_sensative' option, I figured it was limited to the module and not Ansible.  But seems that was not correct.
I will have to break out the multiple user creation for Cassandra into multiple tasks to ensure the credentials are not exposed or use the 'no_log' option.
Thanks
